packages/utils/src/carequality/cq-monthly-stats/shared.ts

```diff
@@ -0,0 +1,323 @@
+  xcaDRStatsByDay: ImplementerStatsByDay
+): MonthlyImplementerStats[] {
+  const monthlyStats: MonthlyImplementerStats[] = [];
+
+  const xcpdDailyAvgsByImplementer = getMonthlyAvgByImplementer(xcpdStatsByDay);
+  const xcaDQDailyAvgsByImplementer = getMonthlyAvgByImplementer(xcaDQStatsByDay);
+  const xcaDRDailyAvgsByImplementer = getMonthlyAvgByImplementer(xcaDRStatsByDay);
+
+  xcpdDailyAvgsByImplementer.forEach(implementerStat => {
+    const { implementerId, implementerName, avgResponseTimeMs } = implementerStat;
+
+    const existingStat = monthlyStats.find(
+      s => s.year === year && s.month === month && s.implementerId === implementerId
+    );
+
+    if (existingStat) {
+      existingStat.xcpdAvgResponseTimeMs = avgResponseTimeMs;
+    } else {
+      monthlyStats.push({
+        year,
+        month,
+        implementerId,
+        implementerName,
+        xcpdAvgResponseTimeMs: avgResponseTimeMs,
+      });
+    }
+  });
+
+  xcaDQDailyAvgsByImplementer.forEach(implementerStat => {
+    const { implementerId, implementerName, avgResponseTimeMs } = implementerStat;
+
+    const existingStat = monthlyStats.find(
+      s => s.year === year && s.month === month && s.implementerId === implementerId
+    );
+
+    if (existingStat) {
+      existingStat.xcaDQAvgResponseTimeMs = avgResponseTimeMs;
+    } else {
+      monthlyStats.push({
+        year,
+        month,
+        implementerId,
+        implementerName,
+        xcaDQAvgResponseTimeMs: avgResponseTimeMs,
+      });
+    }
+  });
+
+  xcaDRDailyAvgsByImplementer.forEach(implementerStat => {
+    const { implementerId, implementerName, avgResponseTimeMs } = implementerStat;
+
+    const existingStat = monthlyStats.find(
+      s => s.year === year && s.month === month && s.implementerId === implementerId
+    );
+
+    if (existingStat) {
+      existingStat.xcaDRAvgResponseTimeMs = avgResponseTimeMs;
+    } else {
+      monthlyStats.push({
+        year,
+        month,
+        implementerId,
+        implementerName,
+        xcaDRAvgResponseTimeMs: avgResponseTimeMs,
+      });
+    }
+  });
+
+  return monthlyStats;
+}
+
+function getMonthlyAvgByImplementer(statsByDay: ImplementerStatsByDay): ImplementerWithAvgResp[] {
+  const monthlyAvgsByImplementer: {
+    [key: string]: {
+      implementerId: string;
+      implementerName: string;
+      avgResponseTimeMs: number[];
+    };
+  } = {};
+
+  Object.values(statsByDay).forEach(stats => {
+    stats.forEach(stat => {
+      const { implementerId, implementerName } = stat;
+      const { gwStats } = stat;
+
+      const avgResponseTimeMs = aggregateGwAvgResponseTime(gwStats);
+
+      monthlyAvgsByImplementer[implementerId] = {
+        implementerId,
+        implementerName,
+        avgResponseTimeMs: [
+          ...(monthlyAvgsByImplementer[implementerId]
+            ? monthlyAvgsByImplementer[implementerId].avgResponseTimeMs
+            : [avgResponseTimeMs]),
+          avgResponseTimeMs,
+        ],
```

Comment:
If there was no item for the current `implementerId`, we're adding `avgResponseTimeMs` twice, right?